x-pack/packages/observability/observability_utils/es/utils/esql_result_to_plain_objects.ts

```diff
@@ -13,7 +13,17 @@ export function esqlResultToPlainObjects<T extends Record<string, any>>(
   return result.values.map((row) => {
     return row.reduce<Record<string, unknown>>((acc, value, index) => {
       const column = result.columns[index];
-      acc[column.name] = value;
+
+      if (!column) {
+        return acc;
+      }
+
+      // Removes the type suffix from the column name
+      const name = column.name.replace(/\.(text|keyword)$/, '');
```

Comment:
Q: does including text or keywords cause any issues, or is it simply a matter of preference?

Reply:
The final object needs to match the type. [entityLatestSchema](https://github.com/crespocarlos/kibana/blob/main/x-pack/packages/kbn-entities-schema/src/schema/entity.ts#L33), for instance. The problem may arise if there is an attribute like `definition_id` in the type while the `column.name` is `definition_id.keyword`. In this case, we'd have an issue.